### DISCUSSION TITLE
Do dapp's requiredNamespaces and wallet's accounts need to fully overlap?

### DISCUSSION THREAD

**Discussion by GrahamMcBain** [WalletConnect/walletconnect-monorepo#1697] [Developer Support]
Hi there! Working on react-celo's package 🙂

I'm trying to make it easy to celo dapp devs to use all three celo networks, so this is how I use signClient.connect:

```
const { uri, approval } = await this.client.connect({
      pairingTopic: this.pairing?.topic,
      requiredNamespaces: {
        eip155: {
          chains: [
            'eip155:44787', // alajores
            'eip155:42220', // celo
            'eip155:62320', // baklava
          ],
          methods: Object.values(SupportedMethods),
          events: ['chainChanged', 'accountsChanged'],
        },
      },
    });

```

And I'm currently testing on my local alfajores wallet to see if everything works by accepting the connection like this:

```
const { topic: _, acknowledged } = await signClient.approve({
        id,
        namespaces: {
          eip155: {
            accounts: [`eip155:${Alfajores.chainId}:${account.address}`],
            methods: requiredNamespaces.eip155.methods,
            events: requiredNamespaces.eip155.events,
          },
        },
      });


```
This doesn't seem to work as I have only an alfajores account and no mainnet/baklava accounts available. Am I missing something obvious here?

- Small note, if I comment out the mainnet and baklava lines, everything works just fine. I'm just wondering how to support all chains from the dapp perspective, knowing some wallets only support mainnet, or only alfajores.

(graham-asks): If you comment out baklava & alajores and just leave mainnet does it work? Do you see what I mean?

(Nicolas Responds): Yes it works
The question is how to make websites using react-celo support wallets that only support a subset of the 3 networks. (eg, valora only supports mainnet, valora-test-app only supports alfajores)

Not sure if my question fully makes sense but yeah 😄

**Comment by roman-utila** [comment]
Same here.

Let's say that my wallet supports only eip155 chains, but not bip122.
And let's say that I want to respond to a request from a dApp, where it asks for approval for both eip155 and bip122.
I noticed that when I respond only with the eip155 in the namespace, I am not able to continue with the flow of the application.
If I remove the bip122 from the approval to session_proposal session, I can't continue.
If I keep bip122 in the approval to session_proposal but pass an empty arrays for the accounts, events and methods, I can't continue.
Only if I keep bip122 and provide a dummy account in the accounts array, as well as providing the requested events and methods I can continue
